### PR TITLE
[API] [Product] [ProductTranslation] Add api configuration and test for adding product and product translations

### PR DESCRIPTION
**Pull request by Tomanhez**
| Q               | A
| --------------- | -----
| Branch?         | api
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| License         | MIT

### CODE REVIEW

File: src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    /**
+     * @When I name it :name in :language
+     * @When I rename it to :name in :language
+     */
+    public function iRenameItToIn(string $name, string $language): void
+    {
+        $data = ['translations' => [$language => ['locale' => $language]]];
+        $data['translations'][$language]['name'] = $name;
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php]:
```suggestion
        $data = [
            'translations' => [
                $language => [
                    'locale' => $language,
                    'name' => $name,
                ],
            ],
        ];
```
```

File: src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    /**
+     * @When I set its slug to :slug
+     * @When I set its slug to :slug in :language
+     * @When I remove its slug
+     */
+    public function iSetItsSlugTo(?string $slug = null, $language = 'en_US'): void
+    {
+        $data = ['translations' => [$language => ['slug' => $slug]]];
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php]: Why is this configuration different than the configuration for the name field?

Comment (Tomanhez) [reply src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php]: Is problem with doubling  locale key when array is merged in addCompoundRequestData(), I think that this method should be refactored 
Screenshot 2020-03-11 at 21 15 36

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/ProductTranslation.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        shortName="ProductTranslation"
+    >
+        <attribute name="validation_groups">sylius</attribute>
+        <collectionOperations />
+        <itemOperations>
+            <itemOperation name="get" />
+        </itemOperations>
+        <property name="id" identifier="true" writable="false" />
+        <property name="name" identifier="false" required="true" />
+        <property name="locale" identifier="false" required="true" />
```
Comment (AdamKasp) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/ProductTranslation.xml]: probably you don't need to add `identifier="false"` everywhere

File: src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php (hunk 1 above)
Comment (lchrusciel) [reply src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php]: What will happen if `'locale' => $language` will be omitted?

File: src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    public function iSpecifyItsCodeAs(string $code = null): void
+    {
+        $this->client->addRequestData('code', $code);
+    }
+
+    /**
+     * @When I name it :name in :language
+     * @When I rename it to :name in :language
+     */
+    public function iRenameItToIn(string $name, string $language): void
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php]:
```suggestion
    public function iRenameItToIn(string $name, string $localeCode): void
```
?

File: src/Sylius/Behat/Resources/config/services/contexts/api.xml [hunk 5]
```diff
@@ -56,6 +56,10 @@
             <argument type="service" id="sylius.behat.shared_storage" />
         </service>
 
+        <service id="sylius.behat.context.api.admin.managing_product" class="Sylius\Behat\Context\Api\Admin\ManagingProductsContext">
```
Comment (GSadee) [inline src/Sylius/Behat/Resources/config/services/contexts/api.xml]:
```suggestion
        <service id="sylius.behat.context.api.admin.managing_products" class="Sylius\Behat\Context\Api\Admin\ManagingProductsContext">
```

File: src/Sylius/Behat/Resources/config/suites.yml [hunk 6]
```diff
@@ -6,6 +6,7 @@ imports:
     - suites/api/currency/managing_currencies.yml
     - suites/api/currency/managing_exchange_rates.yml
     - suites/api/payment/managing_payments.yml
+    - suites/api/product/managing_products.yml
```
Comment (GSadee) [inline src/Sylius/Behat/Resources/config/suites.yml]: Wrong order

File: src/Sylius/Bundle/ApiBundle/Resources/config/serializer/ProductTranslation.xml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+        <attribute name="locale">
+            <group>product:create</group>
+            <group>product:update</group>
+        </attribute>
+        <attribute name="slug">
+            <group>product:create</group>
+            <group>product:read</group>
+        </attribute>
+    </class>
+
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Resources/config/serializer/ProductTranslation.xml]: Unnecessary blank line

File: src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php (hunk 1 above)
Comment (Tomanhez) [reply src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php]: Not working

File: src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+     * @When I set its slug to :slug
+     * @When I set its slug to :slug in :localeCode
+     * @When I remove its slug
+     */
+    public function iSetItsSlugTo(?string $slug = null, $localeCode = 'en_US'): void
+    {
+        $data = [
+            'translations' => [
+                $localeCode => [
+                    'slug' => $slug,
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php]: This should work now:
```suggestion
                    'locale' => $localeCode,
                    'slug' => $slug,
```

File: src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+
+    /**
+     * @Then I should see the product :productName in the list
+     * @Then the product :productName should appear in the store
+     * @Then the product :productName should be in the shop
+     * @Then this product should still be named :productName
+     */
+    public function theProductShouldAppearInTheShop(string $productName): void
+    {
+        $this->client->index();
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php]:
```suggestion
        $response = $this->client->index();
```

File: src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+
+        Assert::true(
+            $this
+                ->responseChecker
+                ->hasItemWithTranslation($this->client->getLastResponse(),'en_US', 'name', $productName)
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php]:
```suggestion
                ->hasItemWithTranslation($response, 'en_US', 'name', $productName)
```

File: src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+        );
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php]: Missing exception message

File: src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+    }
+
+    /**
+     * @Then I should be notified that it has been successfully created
+     */
+    public function iShouldBeNotifiedThatItHasBeenSuccessfullyCreated(): void
+    {
+        Assert::true($this->responseChecker->isCreationSuccessful($this->client->getLastResponse()));
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Api/Admin/ManagingProductsContext.php]: Missing exception message

**Comment by lchrusciel** [comment]
Thank you, Tomasz! :1st_place_medal:

**Comment by lchrusciel** [comment]
Part of #11250

**Pull request merged**